src/documentation/dokumentacja_wstepna.md

```diff
@@ -35,3 +35,70 @@ Dla uproszczenia obliczeń, powyższe równanie można zapisać w sposób macier
+
+### Agent globalny
+Agent globalny odpowiedzialny jest za przydzielanie zasobów chmury w taki sposób, aby
+zoptymalizować zużycie energii, a tym samym koszty utrzymania chmury. Optymalizacja odbywa się poprzez
+utrzymanie właściwego rozkładu maszyn wirtualnych na maszynach fizycznych oraz propagowanie zadań 
+od użytkowników na ich maszyny wirtualne.
+
+Agent globalny redukuje liczbę uruchomionych (a zatem pobierających energię) maszyn fizycznych poprzez 
+redukcję aktywnych maszyn wirtualnych i ich migracji. Dokonuje tego poprzez szybkie zwalnianie zasobów
+wykorzystywanych przez konkretną maszynę wirtualną, gdy skończy ona wykonywać zadania. Zwolnione zasoby
+mogą zostać przydzielone innej maszynie wirtualnej, bez potrzeby alokowania ich na kolejnej maszynie 
+fizycznej. Agent globalny pozostaje w ciągłej komunikacji z hipernadzorcami maszyn fizycznych, dzięki 
+czemu jest w stanie dostarczać użytkownikom żądane zasoby i zwalniać je, gdy tylko jest to możliwe.
+
+W sytuacji, gdy hipernadzorca zużywa za dużo zasobów powiadamia on o tym globalnego agenta. Globalny agent 
+następnie wybiera maszynę wirtualną, która najbardziej wpływa na przeciążenie maszyny fizycznej (np. tę, która 
+zużywa najwięcej przeciążonego zasobu) i dokonuje jej migracji do innej maszyny fizycznej.
+
+Jeżeli natomiast zajdzie sytuacja, w której maszyna fizyczna jest obciążona w małym stopniu, a w systemie 
+znajdują się inne maszyny, na których mogłyby działać uruchomione na niej maszyny wirtualne, hipernadzorca
+również powiadamia globalnego agenta. Globalny agent migruje wtedy wszystkie maszyny wirtualne ze wskazanej
+maszyny fizycznej do innych maszyn, redukując w ten sposób liczbę pracujących maszyn o 1. [1]
+
+Redukcja uruchomionych maszyn fizycznych następuje również poprzez przyjęcie konkretnej strategii alokacji
+zasobów dla nowych maszyn wirtualnych. Proponujemy przyjęcie strategii podobnej do [2] w której wybieramy zasób, na który nowa 
+maszyna ma największe zapotrzebowanie, następnie spośród pracujących maszyn fizycznych wybieramy tę, która posiada go najmniej, ale jest w stanie zapewnić
+wystarczająco dużo zasobów nowej maszynie wirtualnej. Jeżeli żadna z pracujących już maszyn fizycznych nie
+jest w stanie uruchomić nowej maszyny wirtualnej, dokonujemy aktywacji kolejnej maszyny fizycznej, przy czym
+wybierana jest ta o najmniejszym zużyciu energii. 
+
+Aby efektywnie zwalniać zasoby maszyn fizycznych, agent globalny zapewnia również strategię przyporządkowywania 
+zadań do maszyn wirtualnych w taki sposób, aby jak najwięcej z nich mogło wykonywać się równolegle. Proponujemy 
+metodę opartą o zasoby, analogiczną do alokacji zasobów dla maszyn wirtualnych, która została opisana powyżej.
+Wybieramy zasób najistotniejszy dla zadania, następnie maszynę wirtualną, która posiada go najmniej, ale jest 
```

Comment:
jak wybieramy ten zasób?

Reply:
globalny agent chyba nie ma danych o zadaniach żeby stwierdzić, który zasób jest najistotniejszy?

Reply:
W zasadzie nic nie stoi na przeszkodzie żeby miał. We wszystkich opracowaniach na jakie się natknęłam ma informacje np o czasie wykonania, więc w zasadzie o potrzebnym RAMie albo ilości rdzeni chyba też mógłby mieć. Niestety za nic nie mogę się dokopać do artykułu w którym czytałam o takim zasobowym podejściu